cmd/api/store.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"database/sql"
+
+	"github.com/gitpods/gitpods/handler"
+	"github.com/gitpods/gitpods/store"
+	"github.com/gorilla/sessions"
+	_ "github.com/lib/pq"
+)
+
+func NewRouterStore(driver string, dsn string, secret []byte) (*handler.RouterStore, error) {
+	cookieStore := sessions.NewFilesystemStore("./dev/sessions/", secret)
+
+	routerStore := handler.RouterStore{
+		CookieStore: cookieStore,
+	}
+
+	if driver == "memory" {
+		usersStore := store.NewUsersInMemory()
+		repositoriesStore := store.NewRepositoriesInMemory(usersStore)
+		usersRepositoriesStore := store.NewUsersRepositoriesInMemory(usersStore, repositoriesStore)
+
+		routerStore.UsersStore = usersStore
+		routerStore.UsersRepositoriesStore = usersRepositoriesStore
+		routerStore.AuthorizeStore = usersStore
+	}
+
+	if driver == "postgres" {
+		db, err := sql.Open("postgres", dsn)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := db.Ping(); err != nil {
```

Comment:
I'm not sure if I should do that at all. There are people that say an application shouldn't crash if there's not database when bootstrapping. It should rather crash when a (http) handler is triggered. Otherwise you can have deadlocks within the microservices talking to each other.
@fortytw2 what do you think?